src/commands.ts

```diff
@@ -505,18 +507,26 @@ export const insertInlineFootnote =
       ])
 
       const backmatter = findChildrenByType(tr.doc, schema.nodes.backmatter)[0]
-
       tr.insert(backmatter.pos + 1, section)
 
-      // inside footnote inside element inside section
-      selectionPos = backmatter.pos + section.nodeSize - 3
+      let footnotePos = 0
+      tr.doc.descendants((n, pos) => {
```

Comment:
Why are you doing this? Do you really need to trace through the whole document searching for the footnote? Since you just inserted the section with that footnote, couldn't you calculate its position using the information you already have? 🤔

Also, If we have any other footnotes in the document (like those for tables), it will return an incorrect selection position.

Reply:
- my understanding is it's not possible for a table footnote to be in the document in the backmatter after normal footnotes.
- I'll try to revisit that approach but for some reason I was unable to find position reliably using basic calculations and had to find the footnotes like that. I assume this is because there is a paragraph inside and to avoid unreliable assumptions about internal positions in the footnotes I would have to check the descendants of the node anyway but maybe not in the context of the entire document.